dpnp/backend/extensions/lapack/gesv.cpp

```diff
@@ -144,6 +144,16 @@ static sycl::event gesv_impl(sycl::queue &exec_q,
         is_exception_caught = true;
         gesv_utils::handle_lapack_exc(exec_q, lda, a, scratchpad_size,
                                       scratchpad, ipiv, e, error_msg);
+    } catch (oneapi::mkl::computation_error const &e) {
+        // TODO: remove this catch when gh-642(oneMath) is fixed
+        // Workaround for oneMath interfaces
+        // oneapi::mkl::computation_error is thrown instead of
+        // oneapi::mkl::lapack::computation_error.
+        if (scratchpad != nullptr)
+            sycl_free_noexcept(scratchpad, exec_q);
+        if (ipiv != nullptr)
+            sycl_free_noexcept(ipiv, exec_q);
+        throw LinAlgError("The input coefficient matrix is singular.");
```

Comment:
I'm curious why is it handled in another way comparing to `oneapi::mkl::lapack::computation_error` above?

Reply:
Above we handle all exceptions in `oneapi::mkl::lapack::exception` class by defining a specific exception using the value of `info()` method. 
In this case if the input matrix is singular, the exception `oneapi::mkl::computation_error` is thrown.

Also `info` method is not available in `oneapi::mkl::exception:` class

Reply:
I see, the main reason is that we can't use `gesv_utils::handle_lapack_exc` here, because `oneapi::mkl::computation_error` doesn't expose `info()` method.